forte/processors/pretrained_encoder_processors.py

```diff
@@ -0,0 +1,93 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+
+import texar.torch as tx
+
+from texar.torch import HParams
+
+from forte.common.resources import Resources
+from forte.data.data_pack import DataPack
+from forte.processors.base import PackProcessor
+from forte.utils.utils import get_class
+
+
+__all__ = [
+    "PretrainedEncoder",
+]
+
+
+class PretrainedEncoder(PackProcessor):
+    r"""A wrapper of Texar pre-trained encoders.
+    """
+    def __init__(self):
+        super().__init__()
+        self.tokenizer = None
+        self.encoder = None
+        self.entry_type = None
+
+    # pylint: disable=unused-argument
+    def initialize(self, resource: Resources, configs: HParams):
+        if configs.pretrained_model_name.startswith('bert'):
+            self.tokenizer = tx.data.BERTTokenizer(
+                pretrained_model_name=configs.pretrained_model_name)
+            self.encoder = tx.modules.BERTEncoder(
+                pretrained_model_name=configs.pretrained_model_name)
+        elif configs.pretrained_model_name.startswith('gpt2'):
+            self.tokenizer = tx.data.GPT2Tokenizer(
+                pretrained_model_name=configs.pretrained_model_name)
+            self.encoder = tx.modules.GPT2Encoder(
+                pretrained_model_name=configs.pretrained_model_name)
+        elif configs.pretrained_model_name.startswith('roberta'):
+            self.tokenizer = tx.data.RoBERTaTokenizer(
+                pretrained_model_name=configs.pretrained_model_name)
+            self.encoder = tx.modules.RoBERTaEncoder(
+                pretrained_model_name=configs.pretrained_model_name)
+        elif configs.pretrained_model_name.startswith('T5'):
+            self.tokenizer = tx.data.T5Tokenizer(
+                pretrained_model_name=configs.pretrained_model_name)
+            self.encoder = tx.modules.T5Encoder(
+                pretrained_model_name=configs.pretrained_model_name)
+        elif configs.pretrained_model_name.startswith('xlnet'):
+            self.tokenizer = tx.data.XLNetTokenizer(
+                pretrained_model_name=configs.pretrained_model_name)
+            self.encoder = tx.modules.XLNetEncoder(
+                pretrained_model_name=configs.pretrained_model_name)
+        else:
+            raise ValueError("Unrecognized pre-trained model name.")
+
+        self.entry_type = get_class(configs.entry_type,
+                                    ['ft.onto.base_ontology'])
```

Comment:
This just support base ontology? We can simply as the config to be the full entry name so we can support any entry in the ontology. Further, next add a check to see if this class is subclassing `enty`

Reply:
In fact, I realize we nee the check if the class is subclassing `Annotation` or not. The classes in `base_ontology` are not always annotations.